iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NetworkImpl.java

```diff
@@ -76,6 +76,10 @@ public int getSwitchCount() {
 
         @Override
         public Bus getBus(String id) {
+            Bus bus = index.get(id, Bus.class);
```

Comment:
Are you sure it works, and you don't have to use ConfiguredBusImpl.class ?

Reply:
Yes it works, I have double checked it using a profiled run (performance gain factor around 2) and a debug session. The returned Bus instance is of class ConfiguredBusImpl, but I don't need to pass the subtype to the get method. Bus is enough to find it.

Reply:
The implementation of `index.get` looks in the map of objects by id (without type) and later checks the type of the result:

```java
        Identifiable<?> obj = get(id);
        if (obj != null && clazz.isAssignableFrom(obj.getClass())) {
            return (T) obj;
        } else {
```
and a `Bus` is assignable from the subtype `ConfiguredBusImpl`.

Reply:
Ok thanks